{
  "comments": [
    {
      "key": {
        "uuid": "6eb21ced_63cd9fbb",
        "filename": "core/java/android/os/ParcelFileDescriptor.java",
        "patchSetId": 1
      },
      "lineNbr": 422,
      "author": {
        "id": 1013508
      },
      "writtenOn": "2018-09-28T18:24:23Z",
      "side": 1,
      "message": "Are you sure these are safe changes to make?  Should they be guarded by target SDK level of the running app?",
      "range": {
        "startLine": 422,
        "startChar": 0,
        "endLine": 422,
        "endChar": 64
      },
      "revId": "f82214432c7edd6319ef18d83dd12c929aae9cbe",
      "serverId": "85c56323-6fa9-3386-8a01-6480fb634889",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "f62ed06f_bdb7818a",
        "filename": "core/java/android/os/ParcelFileDescriptor.java",
        "patchSetId": 1
      },
      "lineNbr": 422,
      "author": {
        "id": 1079148
      },
      "writtenOn": "2018-09-28T19:38:10Z",
      "side": 1,
      "message": "I don\u0027t think there\u0027s a reasonable way to guard this behind target SDK level, the existing implementations are just broken if the Socket/DatagramSocket get GCed before your PFD. I don\u0027t think it\u0027s very likely that anyone is relying on the existing behavior (especially since it directly contradicts the documentation).\n\nThat being said, dup throws IOException and this has no exception specifier, so it doesn\u0027t compile. Presumably we can\u0027t add exception specifiers to existing APIs? Should I rethrow any exceptions that happen in a RuntimeException or something?",
      "parentUuid": "6eb21ced_63cd9fbb",
      "range": {
        "startLine": 422,
        "startChar": 0,
        "endLine": 422,
        "endChar": 64
      },
      "revId": "f82214432c7edd6319ef18d83dd12c929aae9cbe",
      "serverId": "85c56323-6fa9-3386-8a01-6480fb634889",
      "unresolved": true
    }
  ]
}